grafana.sh

```diff
@@ -279,14 +279,59 @@ set_vcap_datasource_prometheus() {
 }
 
 
+set_vcap_datasource_alertmanager() {
+    local datasource="${1}"
+
+    local label=$(jq -r '.label' <<<"${datasource}")
+    local name=$(jq -r '.name' <<<"${datasource}")
+    local user=$(jq -r '.credentials.prometheus.user | select (.!=null)' <<<"${datasource}")
+    local pass=$(jq -r '.credentials.prometheus.password | select (.!=null)' <<<"${datasource}")
+    local url=$(jq -r '.credentials.alertmanager.url' <<<"${datasource}")
+    local auth="true"
+
+    [[ -z "${user}" ]] && auth="false"
+
+    set +e
+    datasources_folder_exists="$(ls "${APP_ROOT}/datasources")"
+    datasources_folder_exists_exit_code=$?
+    if [[ ${datasources_folder_exists_exit_code} -eq 1 ]]
+    then
+        # Only create datasources directory if it is not already created in set_vcap_datasource_prometheus()
+        mkdir -p "${APP_ROOT}/datasources"
+    fi
+
+    # Be careful, this is a HERE doc with tabs indentation!!
+    cat <<-EOF > "${APP_ROOT}/datasources/${HOME_ORG_ID}-${name}-alertmanager.yml"
+	apiVersion: 1
+	
+	# list of datasources to insert/update depending
+	# what's available in the database
+	datasources:
+	- name: ${name} AlertManager
+	  type: camptocamp-prometheus-alertmanager-datasource
+	  access: proxy
+	  orgId: ${HOME_ORG_ID}
+	  url: "${url}"
+	  basicAuth: ${auth}
+	  basicAuthUser: ${user}
+	  secureJsonData:
+	    basicAuthPassword: ${pass}
+	  withCredentials: false
+	  isDefault: false
+	  editable: true
+	EOF
+}
+
 set_datasources() {
     local datasource
 
     datasource=$(get_binding_service "${DATASOURCE_BINDING_NAME}")
     [[ -z "${datasource}" ]] && datasource=$(get_prometheus_vcap_service)
     if [[ -n "${datasource}" ]]
     then
+        echo "camptocamp-prometheus-alertmanager-datasource 0.0.7" > ${GRAFANA_CFG_PLUGINS}
```

Comment:
We should use `>>` here, to avoid overriding the whole plugins file

Reply:
This is a good point.

Through speaking to Jose, he proposed a whole variety of changes across various files (instead of hard-coding 0.0.7). I didn't agree with 100% of his proposals, but some of it I agreed with.

So this is where the bulk of the refactoring work is taking place.

I'll make a new commit when I've refactored.

Reply:
I've refactored and committed new changes regarding the hard-coded version number.